src/main/java/com/gmail/goosius/siegewar/integration/maptowny/MapTownyIntegration.java

```diff
@@ -0,0 +1,219 @@
+package com.gmail.goosius.siegewar.integration.maptowny;
+
+import com.gmail.goosius.siegewar.SiegeController;
+import com.gmail.goosius.siegewar.SiegeWar;
+import com.gmail.goosius.siegewar.enums.SiegeSide;
+import com.gmail.goosius.siegewar.objects.BattleSession;
+import com.gmail.goosius.siegewar.objects.Siege;
+import com.gmail.goosius.siegewar.settings.Settings;
+import com.palmergames.bukkit.towny.TownyEconomyHandler;
+import com.palmergames.bukkit.towny.TownyUniverse;
+import com.palmergames.bukkit.towny.object.TownyWorld;
+import com.palmergames.bukkit.towny.object.Translation;
+import com.palmergames.util.StringMgmt;
+import me.silverwolfg11.maptowny.MapTownyPlugin;
+import me.silverwolfg11.maptowny.objects.LayerOptions;
+import me.silverwolfg11.maptowny.objects.MarkerOptions;
+import me.silverwolfg11.maptowny.objects.Point2D;
+import me.silverwolfg11.maptowny.platform.MapLayer;
+import me.silverwolfg11.maptowny.platform.MapPlatform;
+import me.silverwolfg11.maptowny.platform.MapWorld;
+import org.apache.commons.lang.WordUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.logging.Level;
+
+public class MapTownyIntegration {
+
+    private final String SIEGE_MARKER_PREFIX = "siegewar_siege_";
+    private final String SIEGE_LAYER_PREFIX = "siegewar_layer_";
+    private final String PEACEFUL_BANNER_ICON_KEY = "siegewar_peaceful";
+    private final String BATTLE_BANNER_ICON_KEY = "siegewar_battle";
+
+    private final MapTownyPlugin mapTowny;
+    private final Map<String, MapLayer> worldLayers = new HashMap<>();
+    private boolean siegesRendered = false;
+
+    public MapTownyIntegration(SiegeWar plugin) {
+        this.mapTowny = (MapTownyPlugin) Bukkit.getPluginManager().getPlugin("MapTowny");
+
+        // Check if map towny found a valid web-map platform
+        if (mapTowny.getPlatform() == null) {
+            return;
+        }
+
+        // Register replacement handler
+        Bukkit.getPluginManager().registerEvents(new MapTownyReplacementsHandler(mapTowny), plugin);
+
+        // Register repeating task that runs every 2 seconds delayed by 15 seconds on start.
+        Bukkit.getScheduler().runTaskTimerAsynchronously(plugin, this::displaySieges, 40L, 300L);
```

Comment:
🚧 
- Can the above repeat time be increased ? E.g. to maybe 10.
- Sw screens are enerally optimised for performance in this way, for example the Sw hud only updates every 20 secs.
- If you wanted to go one step further for perf you could trigger the map change by listening for the Towny Short Tick, but the latter is only an idea not a request.

Reply:
Done! In the latest push, the map change is now triggered by the NewShortTimeEvent.